week1/lc378_kthSmallest.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * @param {number[][]} matrix
+ * @param {number} k
+ * @return {number}
+ */
+const kthSmallest = function(matrix, k) {
+    let start = matrix[0][0];
+    let end = matrix[matrix.length - 1][matrix.length - 1];
+
+    while (start + 1 < end) {
+        let mid = start + ((end - start) >> 1);
+
+        if (countLessAndEqualT(matrix, mid) < k) {
+            start = mid;
+        }
+        else {
+            end = mid;
+        }
+    }
```

Comment:
I don't know if my understanding is right. There will be three situations when we execute countLessAndEqualT(matrix, start).
case 1: (countLessAndEqualT(matrix, start) == k)
We can return start.

case 2: (countLessAndEqualT(matrix, start) < k)
It means start is not a possible solution, because we need to find the Kth smallest number. So return end.

case 3: (countLessAndEqualT(matrix, start) > k)
We already got count greater than k when we execute this function with start as argument. So if we execute this function with end as argument, the count is definitely going to be greater than or equal to current count. So we return start.

Reply:
We use `countLessAndEqualT` everytime we calculate a new `mid`. 

It is used to count how many items in the matrix that the value is less or equal to `T`.

Let's say given a matrix, and given a target (7)
[1,2,3][4,5,6][7,8,9]
We need to find how many items that the value is less and equal to 7. 

We run countLessAndEqualT(matrix, 7) => it returns 7.